srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/NewRuleController.java

```diff
@@ -0,0 +1,136 @@
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.ModelAndView;
+
+import com.google.common.primitives.Ints;
+
+/**
+ * <p>Web MVC controller for creating a new rule.</p>
+ */
+@Controller
+public class NewRuleController
+{
+    public static final String NEW_RULE_URL = "/admin/newRule";
+    
+    private static final Logger fLogger = LoggerFactory.getLogger(NewDiscreteNumericalVarController.class);
+    /**
+     * The model attribute key of the {@link Rule} object.
+     */
+    protected static final String ATTRIBUTE_RULE = "rule";
+    
+    private final AdminService fAdminService;
+    
+    /**
+     * Constructs an instance.
+     * @param adminService the AdminService to use for operations
+     * @throws NullPointerException if adminService is null
+     */
+    @Inject
+    public NewRuleController(final AdminService adminService)
+    {
+        fAdminService = Objects.requireNonNull(adminService);
+    }
+    
+    @ModelAttribute(ATTRIBUTE_RULE)
+    private EditRule createEditRule()
+    {
+        fLogger.trace("Creating EditDiscreteNumericalVar.");
+        return new EditRule(fAdminService);
+    }
+    
+    @RequestMapping(value = NEW_RULE_URL, method = RequestMethod.GET)
+    public ModelAndView displayForm(
+            @ModelAttribute(ATTRIBUTE_RULE) final EditRule editRule) throws InvalidIdentifierException
+    {
+        final ModelAndView mav = new ModelAndView(Views.EDIT_RULE);
+        // Provide the URL to save the rule to the view.
+        mav.addObject("SAVE_URL", NEW_RULE_URL);
+        // Note: "rule" is already in the model via createEditRule().
+        mav.addObject("isNewRule", true);
+        // A map of the variable's key to its respective summary
+        final Map<String, VariableSummary> variableSummaries = new HashMap<String, VariableSummary>();
+        for(final ValueMatcherBuilder matcher: editRule.getMatchers())
+        {
+            final Variable var = fAdminService.getVariable(matcher.getVariableKey());
+            variableSummaries.put(matcher.getVariableKey(), VariableSummary.fromVariable(var));
+        }
+        mav.addObject("variableSummaries", variableSummaries);
+        mav.addObject("allVariables", fAdminService.getAllVariables());
+        return mav;
+    }
+    
+    @RequestMapping(value = NEW_RULE_URL, method = RequestMethod.POST)
+    public ModelAndView saveRule(
+            @ModelAttribute(ATTRIBUTE_RULE) final EditRule editRule,
+            final BindingResult bindingResult,
+            @RequestParam(value = "submitButton") final String submitString) throws InvalidIdentifierException
+    {
+        if(submitString.equals("newMatcher"))
+        {
+            // Reload the page with the new matcher builder inside of the edit rule
+            editRule.getMatchers().add(
+                    new ValueMatcherBuilder().setVariableKey(editRule.getNewVariableKey()));
```

Comment:
Since a ValueMatcherBuilder is always tied to a Variable, why not just take the variable key in the constructor?